skbio/alignment/_tabular_msa.py

```diff
@@ -715,8 +792,92 @@ def reindex(self, key=None, keys=None):
             keys_.flags.writeable = False
 
         self._keys = keys_
+        self._minter = minter
 
     @experimental(as_of='0.4.0-dev')
+    def append(self, sequence, minter=None):
```

Comment:
Yep that's right: a minter/key is required if there are keys on the MSA. I just followed up on #1125.